polygon/bridge/reader.go

```diff
@@ -0,0 +1,94 @@
+package bridge
+
+import (
+	"context"
+	"fmt"
+
+	libcommon "github.com/erigontech/erigon-lib/common"
+	"github.com/erigontech/erigon-lib/log/v3"
+	"github.com/erigontech/erigon/common/u256"
+	"github.com/erigontech/erigon/core"
+	"github.com/erigontech/erigon/core/state"
+	"github.com/erigontech/erigon/core/types"
+
+	"github.com/erigontech/erigon-lib/kv"
+	"github.com/erigontech/erigon/polygon/polygoncommon"
+)
+
+type Reader struct {
+	store Store
+	log   log.Logger
+}
+
+func AssembleReader(ctx context.Context, dataDir string, logger log.Logger) (*Reader, error) {
+	bridgeDB := polygoncommon.NewDatabase(dataDir, kv.PolygonBridgeDB, databaseTablesCfg, logger)
+	bridgeStore := NewStore(bridgeDB)
+
+	err := bridgeStore.Prepare(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewReader(bridgeStore, logger), nil
+}
+
+func NewReader(store Store, log log.Logger) *Reader {
+	return &Reader{
+		store: store,
+		log:   log,
+	}
+}
+
+// Events returns all sync events at blockNum
+func (r *Reader) Events(ctx context.Context, blockNum uint64) ([]*types.Message, error) {
```

Comment:
is the logic here duplicate of bridge.Events?
would be good to remove this duplication, few options:
-maybe we completely remove bridge.Events from bridge.Service and pass bridge.Reader to rpcdaemon and bor consensus engine?
-maybe bridge.Events() re-uses the bridge.Reader.Events()?